src/components/common/CheckboxComponent.jsx

```diff
@@ -1,46 +1,30 @@
-import React, { Component } from 'react';
+import React from 'react';
 import PropTypes from 'prop-types';
 
-class CheckboxComponent extends Component {
-  state = {
-    isChecked: false
-  };
+import '../../_css/CheckboxComponent.css';
 
-  toggleCheckboxChange = () => {
-    const { handleCheckboxChange, label, name } = this.props;
 
-    this.setState(({ isChecked }) => ({
-      isChecked: !isChecked
-    }));
+const CheckboxComponent = (props) => {
+  const { isChecked, label, name, handleCheckboxChange } = props;
 
-    handleCheckboxChange(label, name);
-  };
+  const checkbox = isChecked
```

Comment:
This ternary here is unnecessary.  You can assign the `isChecked` value to the `checked` attribute directly, while giving it a default value of `false`. I.e.
```
const Checkbox = ({ isChecked = false, label, name, onCheckboxChange }) => (
  <div className="field">
    <div className="ui checkbox">
       <input
        type="checkbox"
        name={name}
        value={label}
        onChange={onCheckboxChange}
        checked={isChecked}
      />
      <label className="check label">{label}</label>
    </div>
  </div>
);
```
By assigning a default value for `isChecked`, you don't have to make the prop required in the `propTypes`